homeassistant/components/supla/cover.py

```diff
@@ -0,0 +1,62 @@
+"""Support for Supla cover - curtains, rollershutters etc."""
+import logging
+from pprint import pformat
+
+from homeassistant.components.cover import (
+    ATTR_POSITION, ENTITY_ID_FORMAT, CoverDevice)
+from homeassistant.components.supla import SuplaChannel
+
+DEPENDENCIES = ['supla']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Supla covers."""
+    if discovery_info is None:
+        return
+
+    _LOGGER.debug('Discovery: %s', pformat(discovery_info))
+
+    add_entities([
+        SuplaCover(device) for device in discovery_info
+    ])
+
+
+class SuplaCover(SuplaChannel, CoverDevice):
+    """Representation of a Supla Cover."""
+
+    def __init__(self, channel_data):
+        """Initialize the Supla device."""
+        super().__init__(channel_data)
+        self.entity_id = ENTITY_ID_FORMAT.format(self.unique_id)
```

Comment:
We don't need to implement entity_id if we implement unique_id. The user will be able to change and set the entity_id easily via frontend.

Reply:
Fixed.